Ecal/src/Ecal/EcalWABRecProcessor.cxx

```diff
@@ -0,0 +1,847 @@
+/**
+ * @file EcalWABRecProcessor.h
+ * @brief Processor that reconstructs important kinematic variables for WAB
+ * studies
+ * @author Sanjit Masanam, UCSB
+ */
+
+#include "Ecal/EcalWABRecProcessor.h"
+
+// LDMX
+#include "DetDescr/SimSpecialID.h"
+#include "Ecal/Event/EcalHit.h"
+#include "Recon/Event/EventConstants.h"
+#include "SimCore/Event/SimParticle.h"
+#include "SimCore/Event/SimTrackerHit.h"
+#include "Tracking/Event/StraightTrack.h"
+
+/*~~~~~~~~~~~*/
+/*   Tools   */
+/*~~~~~~~~~~~*/
+#include "Eigen/Dense"
+#include "Tools/AnalysisUtils.h"
+
+// C++
+#include <stdlib.h>
+
+#include <algorithm>
+#include <cmath>
+#include <fstream>
+#include <iomanip>
+#include <numbers>  // For std::numbers::pi
+#include <numeric>
+#include <tuple>
+#include <vector>
+
+// ROOT (MIP tracking)
+#include "TDecompSVD.h"
+#include "TMatrixD.h"
+#include "TVector3.h"
```

Comment:
The includes (that you actually use, which is for sure not all these) should be in the header file